crates/sui-framework/packages/sui-system/tests/validator_set_tests.move

```diff
@@ -313,89 +307,265 @@ module sui_system::validator_set_tests {
     }
 
     #[test]
-    fun test_low_stake_departure() {
+    #[expected_failure(abort_code = validator_set::ENotAValidator)]
+    fun test_request_add_then_pull_stake() {
+        // get enough stake to be added, then pull it before the epoch change
         let mut scenario_val = test_scenario::begin(@0x0);
         let scenario = &mut scenario_val;
-        let ctx = scenario.ctx();
-        // Create 4 validators.
-        let v1 = create_validator(@0x1, 1, 1, true, ctx); // 100 SUI of stake
-        let v2 = create_validator(@0x2, 4, 1, true, ctx); // 400 SUI of stake
-        let v3 = create_validator(@0x3, 10, 1, true, ctx); // 1000 SUI of stake
-        let v4 = create_validator(@0x4, 4, 1, true, ctx); // 400 SUI of stake
 
-        let mut validator_set = validator_set::new(vector[v1, v2, v3, v4], ctx);
+        // Create 2 validators, with stake 9_997 and stake 3
+        let validator1 = create_validator_with_initial_stake(@0x1, 1, 9_997, true, scenario.ctx());
+        let new_v = create_validator_with_initial_stake(@0xA, 0, 0, false, scenario.ctx());
+        // Create a validator set with only the first validator in it
+        let mut validator_set = validator_set::new(vector[validator1], scenario.ctx());
+        advance_epoch_with_dummy_rewards(&mut validator_set, scenario);
+        skip_to_min_stake_v2_final_thresholds(scenario);
+        scenario.next_tx(@0xA);
+
+        validator_set.request_add_validator_candidate(new_v, scenario.ctx());
+        let stake = validator_set.request_add_stake(@0xA, balance::create_for_testing(3 * MIST_PER_SUI), scenario.ctx());
+        validator_set.request_add_validator(scenario.ctx()); // can be admitted
+        let bal = validator_set.request_withdraw_stake(stake, scenario.ctx()); // should fail here with ENotAValidator
+        advance_epoch_with_dummy_rewards(&mut validator_set, scenario);
```

Comment:
For clarity when reading tests, I suggest omitting the cleanup after the expected failure expression and replace it with `abort`.